recipes/cyclonedds-cxx/all/conanfile.py

```diff
@@ -0,0 +1,163 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMake, CMakeDeps, cmake_layout
+from conan.tools.files import copy, export_conandata_patches, get, rm, rmdir, replace_in_file
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.61.0"
+
+class CycloneDDSCXXConan(ConanFile):
+    name = "cyclonedds-cxx"
+    license = "EPL-2.0"
+    homepage = "https://cyclonedds.io/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Eclipse Cyclone DDS C++ Binding- An implementation"\
+                  " of the OMG Data Distribution Service (DDS) specification"
+    topics = ("dds", "ipc", "ros", "middleware")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    short_paths = True
+
+    @property
+    def _min_cppstd(self):
+        return "17"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "Visual Studio": "16",
+            "msvc": "192",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+
+    def _has_idlc(self, info=False):
+        # don't build idllib when it makes little sense or not supported
+        host_os = self.info.settings.os if info else self.settings.os
+        return host_os not in ["Android", "iOS", "watchOS", "tvOS", "Neutrino"]
+
+    def export_sources(self):
+        copy(self, os.path.join("cmake", "CycloneDDS-CXX_idlcxx.cmake"), self.recipe_folder, self.export_sources_folder)
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        # Use the corresponding version of cyclonedds
+        self.requires("cyclonedds/{}".format(self.version), transitive_headers=True)
```

Comment:
Which header in cyclonedds-cxx uses cyclonedds header as public? It would be great a comment like:

`#INFO <cyclonedds-cxx/foobar.h>:95 uses <cyclonedds/bar.h>` 

So we can track it easier, as transitive_headers should be used only when a public header includes a dependency.

Reply:
Documented some of the dependencies in a comment.